### PR TITLE
More Asphalt 5 docs updates

### PR DESCRIPTION
**Pull request by agronholm**
## Checklist

If this is a user-facing code change, like a bugfix or a new feature, please ensure that
you've fulfilled the following conditions (where applicable):

- [ ] You've added tests (in `tests/`) added which would fail without your patch
- [X] You've updated the documentation (in `docs/`, in case of behavior changes or new
features)
- [ ] You've added a new changelog entry (in `docs/versionhistory.rst`).

If this is a trivial change, like a typo fix or a code reformatting, then you can ignore
these instructions.

### Updating the changelog

If there are no entries after the last release, use `**UNRELEASED**` as the version.
If, say, your patch fixes issue <span>#</span>123, the entry should look like this:

```
- Fix big bad boo-boo in task groups
  (`#123 <https://github.com/asphalt-framework/asphalt/issues/123>`_; PR by @yourgithubaccount)
```

If there's no issue linked, just link to your pull request instead by updating the
changelog after you've created the PR.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@
   :target: https://asphalt.readthedocs.io/en/latest/?badge=latest
   :alt: Documentation Status
 
-Asphalt is an asyncio_ based microframework for network oriented applications.
+Asphalt is an asynchronous microframework for network oriented applications.
```
Comment (davidbrochart) [inline README.rst:11]: Do we still need to mention "network oriented applications"? AFAIK Asphalt is "just" an asynchronous microframework.

File: docs/userguide/concurrency.rst [hunk 2]
```diff
@@ -1,121 +1,110 @@
...
 
-Consider a coroutine function that reads the contents of a certain file and then sends them over a
-network connection. While you might get away with reading the file in the event loop thread,
-consider what happens if the disk has to spin up from idle state or the file is located on a slow
-(or temporarily inaccessible) network drive. The whole event loop will then be blocked for who
-knows how long.
+* ``cancel`` (the default): cancel the task and wait for it to finish
+* a callable: run the given callable at teardown to trigger the task to shut itself down
+  (and wait for the task to finish on its own)
+* ``None``: to do nothing and just wait for the task to finish on its own
```
Comment (davidbrochart) [inline docs/userguide/concurrency.rst:41]:
```suggestion
* ``None``: do nothing and just wait for the task to finish on its own
```

File: docs/userguide/concurrency.rst [hunk 3]
```diff
@@ -1,121 +1,110 @@
...
-        contents = await ctx.call_in_executor(Path('file.txt').read_bytes)
-        await connection.send(contents)
+Background task factories are meant to be used for dynamically spawning background tasks
+on demand. A typical example would be a web app request handler that needs to send an
+email, but wants to return a response to the end user straight away. The background
+task will thus outlive the task that was spawned to handle the request.
 
-You can also opt to execute entire blocks with a thread pool executor by using
-:func:`~asphalt.core.context.threadpool`::
+In contrast to service tasks, any running background tasks will block the teardown of
```
Comment (davidbrochart) [inline docs/userguide/concurrency.rst:56]:
```suggestion
In contrast to service tasks, any running background task will block the teardown of
```

File: docs/userguide/concurrency.rst [hunk 4]
```diff
@@ -1,121 +1,110 @@
...
+    :doc:`anyio:threads` section in AnyIO's documentation.
 
-    file_ops = ThreadPoolExecutor(5)  # max 5 worker threads for file operations
-    ctx.add_resource(file_ops, 'file_ops', types=[Executor])
+Configuring the maximum amount of worker threads
+++++++++++++++++++++++++++++++++++++++++++++++++
 
-You can then use this executor resource by its name::
+The configuration option ``max_threads`` sets the limit on how many worker threads will
+at most be used with :func:`anyio.to_thread.run_sync` if not explicit capacity limiter
```
Comment (davidbrochart) [inline docs/userguide/concurrency.rst:94]:
```suggestion
at most be used with :func:`anyio.to_thread.run_sync` if no explicit capacity limiter
```

File: README.rst (hunk 1 above)
Comment (agronholm) [reply README.rst:11]: How would you prefer to word this?

Comment (davidbrochart) [reply README.rst:11]: Maybe:
```suggestion
Asphalt is a microframework for building applications and orchestrating asynchronous services.
```

Comment (davidbrochart) [reply README.rst:11]: > But how about "Asphalt is an asynchronous application microframework"?

That works too, I'm just thinking it's too bad we don't emphasize what it's good at: starting services and tearing them down, in a "safe" way.
What do you you think about replacing "orchestrating" by "managing"?

Comment (agronholm) [reply README.rst:11]: Trying to shoehorn its complex capabilities into a single sentence is probably counterproductive. If we're to advertise those, it should be done in more detail later in the document.

Comment (davidbrochart) [reply README.rst:11]: OK, then "Asphalt is an asynchronous application microframework" is a good one-sentence description :+1:

Comment (agronholm) [reply README.rst]: Updated.

**Review by davidbrochart — APPROVED** [review]

**Pull request merged**